### DISCUSSION TITLE
Extracting subgraph from Heterograph

### DISCUSSION THREAD

**Discussion by GianlucaDeStefano** [pyg-team/pytorch_geometric#6497] [Q&A]
Hi all, 

My task is simple; given an edge in an heterograph, I want to extract its neighborhood graph up to a certain number of hops. 

Looking at the documentation, I have yet to find a good way to do this. 
I found the utility [k_hop_subgraph](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.k_hop_subgraph), but unfortunately, it does not support HeteroGraphs. 

Exploring the code of Pyg found that /sampler/__init__.py  contains the function  "hetero_neighbor_sample" that is, however, undocumented. Can someone please clarify to me the difference between the parameters: 
```
    rowptr_dict: Dict[EdgeType, Tensor],
    col_dict: Dict[EdgeType, Tensor]
```
In practice, what should they contain?
is there maybe a better function to use that I have missed?

**Comment by wsad1** [comment]
A hacky way of doing this would be to use the [LinkNeighborLoader](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/loader/link_neighbor_loader.html#LinkNeighborLoader). 
```
loader = LinkNeighborLoader( data = your_data, num_neighbors = [-1]*num_hops, edge_label_index = ((edge_type), edge_index))
subgraph = next(iter(loader))
```
Given this is hacky. It would be nice to have `k_hop_subgraph` support heterogenous graphs.

**Reply by rusty1s** [reply]
Second this. Currently, the most efficient way is to utilize the `NeighborLoader` classes for this. If you want to start sampling from nodes rather than links, you can also use `NeighborLoader` rather than `LinkNeighborLoader`.
